### PR TITLE
added optional marker icon option for atlas

### PR DESCRIPTION
**Pull request by phodonou**
Description:

Developer now has option to add a MapIcon object to Atlas for customized marker icon.

### CODE REVIEW

**Review by felangel — DISMISSED** [review]

File: packages/atlas/lib/src/map_icon.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import 'package:meta/meta.dart';
+
+/// Optional Marker icon.
+class MapIcon {
```
Comment (felangel) [inline packages/atlas/lib/src/map_icon.dart]:
```suggestion
class MarkerIcon {
```

File: packages/atlas/test/marker_icon_test.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:atlas/atlas.dart';
+
+main() {
+  group('Map Icon', () {
```
Comment (felangel) [inline packages/atlas/test/marker_icon_test.dart]:
```suggestion
  group('Marker Icon', () {
```

**Review by felangel — DISMISSED** [review]

File: packages/atlas/lib/src/marker_icon.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import 'package:meta/meta.dart';
+
+/// Optional Marker icon.
+class MarkerIcon {
+  /// The [assetName] argument must not be null. It should name the main asset
+  /// from the set of images to choose from. File should be of type png
+  final String assetName;
+
+  /// The device pixel ratio where the image will be shown
+  final double devicePixelRatio;
```
Comment (jorgecoca) [inline packages/atlas/lib/src/marker_icon.dart]: A few things here:

- I would rename the variable to just `aspectRatio`
- I would update the document to be more specific (also, it does not matter where the image will be shown)
- In the constructor, make it optional and default to 1

**Review by jorgecoca — CHANGES_REQUESTED** [review]

File: packages/atlas/lib/src/marker_icon.dart (hunk 3 above)
Comment (phodonou) [reply packages/atlas/lib/src/marker_icon.dart]: i just completely removed devicePixelRatio. The developer can just pass in an image of any size

File: packages/atlas/lib/src/marker.dart [hunk 4]
```diff
@@ -9,11 +9,18 @@ class Marker {
   /// The location where the `Marker` is drawn is represented as `LatLng`.
   final LatLng position;
 
+  /// Optional MapIcon used to replace default icon
```
Comment (noahmateen) [inline packages/atlas/lib/src/marker.dart]:
```suggestion
  /// Optional MarkerIcon used to replace default icon
```

**Review by noahmateen — DISMISSED** [review]

File: packages/atlas/lib/src/marker.dart (hunk 4 above)
Comment (phodonou) [reply packages/atlas/lib/src/marker.dart]: good catch

**Review by tchabot22 — APPROVED** [review]

**Review by felangel — APPROVED** [review]

**Review by noahmateen — APPROVED** [review]

**Review by jorgecoca — APPROVED** [review]

**Pull request merged**